### Post-alignment processing of TopHat2 .bam files


[Check out TopHat2 on Galaxy](https://usegalaxy.org/)
<br />
<br />

---


.bam files were generated by aligning RNA-seq reads to a *Lotus japonicus* reference genome sourced from [Kazusa](http://www.kazusa.or.jp/lotus/). Alignment was informed by using a gene annotation file (.gtf) from Kazusa. 
<br />

##### Sort .bam file
```
samtools sort -n Galaxy_accepted_hits.bam Galaxy_accepted_hits_sorted.bam
```
##### Convert .bam to .sam 

Commands were run in the Ubuntu 14.04 shell
```
samtools view -h Galaxy_accepted_hits_sorted.bam > Galaxy_accepted_hits_sorted.sam
```
<br />
 

<br />
##### Generate a counts file in HTSeq 
```
htseq-count --stranded=no --mode=intersection-nonempty -t exon -i gene_id Galaxy_accepted_hits_sorted.sam ~/path/TopHat2.1.0/Lj2.5_gene_models.gtf > LFY.genes.txt
```
<br />

*see Trinity instructions to build expression matrix and calculate differential expression and .R file in /TopHat to normalize counts.
